beacon-chain/cache/depositsnapshot/deposit_tree_test.go

```diff
@@ -0,0 +1,21 @@
+package depositsnapshot
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/prysmaticlabs/prysm/v3/testing/assert"
+)
+
+func Test_AddToEmptyDepositTree(t *testing.T) {
+	dt := &DepositTree{
+		tree:                    &ZeroNode{depth: DepositContractDepth},
+		depositCount:            0,
+		finalizedExecutionblock: [32]byte{},
+	}
+	newDt := NewDepositTree()
+	assert.Equal(t, dt.tree.GetRoot(), newDt.tree.GetRoot())
+	err := newDt.AddDeposit(hexString(t, fmt.Sprintf("%064d", 1)), uint64(0))
+	assert.NoError(t, err)
+	assert.NotEqual(t, dt.tree.GetRoot(), newDt.tree.GetRoot())
```

Comment:
@prestonvanloon something is off, my root is always the same despite adding deposits. 
If you could have a look.

Reply:
Solved it, I missed a break when implementing the algorithm in Go.